packages/api/src/router/user.ts

```diff
@@ -20,6 +20,16 @@ export const profileRouter = createTRPCRouter({
         .returning()
         .then(takeFirst),
     ),
+  updateUsername: protectedProcedure
+    .input(z.object({ username: z.string().min(3).max(30) }))
+    .mutation(({ ctx, input }) =>
+      ctx.db
+        .update(user)
+        .set({ username: input.username })
+        .where(eq(user.id, ctx.session.user.id))
+        .returning()
+        .then(takeFirst),
+    ),
```

Comment:
this should be under the same update function. it doesn't need to be its own endpoint